src/custom/state/claim/hooks/index.ts

```diff
@@ -94,3 +114,160 @@ export function useUserUnclaimedAmount(account: string | null | undefined): Curr
+            })
+            return response.hash
+          })
+      )
+    },
+    [account, addTransaction, chainId, claims, connectedAccount, vCowContract, vCowToken]
+  )
+
+  return { claimCallback }
+}
+
+type GetClaimManyArgsParams = {
+  claimInput: ClaimInput[]
+  claims: UserClaims
+  account: string
+  connectedAccount: string
+}
+
+function _getClaimManyArgs({ claimInput, claims, account, connectedAccount }: GetClaimManyArgsParams): {
+  args: [number[], ClaimType[], string[], string[], string[], string[][], string[]] | undefined
+  totalClaimedAmount: JSBI
+  value: string | undefined
+} {
+  const indices: number[] = []
+  const claimTypes: ClaimType[] = []
+  const claimants: string[] = []
+  const claimableAmounts: string[] = []
+  const claimedAmounts: string[] = []
+  const merkleProofs: string[][] = []
+  const sendEth: string[] = []
+
+  let totalClaimedAmount = JSBI.BigInt('0')
+  let totalValue = JSBI.BigInt('0')
+
+  // Creating a map for faster access when checking what's being claimed
+  const claimsMap = claims.reduce<Record<number, UserClaimData>>((acc, claim) => {
+    acc[claim.index] = claim
+    return acc
+  }, {})
+
+  claimInput.forEach((input) => {
+    const claim = claimsMap[input.index]
+
+    // It can be that the index being passed is already claimed or belongs to another account
+    // Thus, it's possible that the returned `args` is "empty"
+    if (claim) {
+      indices.push(claim.index)
+      // always the same
+      claimants.push(account)
+      // always the max available
+      claimableAmounts.push(claim.amount)
+
+      claimTypes.push(claim.type)
+      // depends on claim type and whether claimed account == connected account
+      const claimedAmount = _getClaimedAmount({ claim, input, account, connectedAccount })
+      claimedAmounts.push(claimedAmount)
+
+      merkleProofs.push(claim.proof)
+      // only used on UserOption, equal to claimedAmount
+      const value = claim.type === ClaimType.UserOption ? claimedAmount : '0'
+      sendEth.push(value) // TODO: verify ETH balance < input.amount ?
+
+      // sum of claimedAmounts for the toast notification
+      totalClaimedAmount = JSBI.add(totalClaimedAmount, JSBI.BigInt(claimedAmount))
+      // sum of Native currency to be used on call options
+      totalValue = JSBI.add(totalValue, JSBI.BigInt(value))
+    }
+  })
+
+  return {
+    args:
+      indices.length > 0
+        ? [indices, claimTypes, claimants, claimableAmounts, claimedAmounts, merkleProofs, sendEth]
+        : undefined,
+    totalClaimedAmount,
+    value: totalValue.toString() === '0' ? undefined : totalValue.toString(),
+  }
+}
+
+type GetClaimedAmountParams = Pick<GetClaimManyArgsParams, 'account' | 'connectedAccount'> & {
+  claim: UserClaimData
+  input: ClaimInput
+}
+
+function _getClaimedAmount({ claim, input, account, connectedAccount }: GetClaimedAmountParams): string {
+  if (
+    // claiming a paid claim on behalf of someone else?
```

Comment:
i would cache these as well with names specific to their state, like `isClaimingForOther = ....`